core/finality-grandpa/src/communication/gossip.rs

```diff
@@ -476,7 +496,29 @@ impl<Block: BlockT> Inner<Block> {
 		}
 	}
 
-	fn consider_vote(&self, round: Round, set_id: SetId) -> Consider {
+	fn consider_vote(&mut self, round: Round, set_id: SetId, msg: &SignedMessage<Block>) -> Consider {
+		let mut tally = self.votes_tally.entry(msg.id.clone()).or_insert(Default::default());
+
+		match &msg.message {
+			PrimaryPropose(_propose) => {
+				tally.handled_primary_proposals += 1;
+			},
+			Prevote(_prevote) => {
+				tally.handled_pre_votes += 1;
```

Comment:
perhaps this and the others should be a saturating addition so we don't overflow somehow?